### ISSUE TITLE
GD *must* throw an error in development if `{:immutable? true}` is supplied and `:id` is nil

### ISSUE THREAD

**Issue by sgrove** [kovasb/gamma-driver#13]

**Comment by sgrove** [comment]
I keep losing time to this.

**Comment by kovasb** [comment]
I don't think this is an issue. The identity of a buffer is modulated by :mapping-fn, which is pluggable; the default behavior is to use :element (bad name, but is the variable the buffer was bound to) which I think is sensible.
